### ISSUE TITLE
React 16 error

### ISSUE THREAD

**Issue by Kolyasya** [alpertuna/react-metismenu#32]
Hello, when I try to use your package with React 16 it has an error:

```
modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37651 Uncaught TypeError: Cannot read property 'oneOfType' of undefined
    at Item.js (modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37651)
    at fileEvaluate (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343)
    at require (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238)
    at Item.js (modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37809)
    at fileEvaluate (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343)
    at require (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238)
    at Container.js (modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37514)
    at fileEvaluate (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343)
    at require (modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238)
    at Container.js (modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37766)
Item.js @ modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37651
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
Item.js @ modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37809
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
Container.js @ modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37514
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
Container.js @ modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37766
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
MetisMenu.js @ modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37236
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
index.js @ modules.js?hash=b4b6fa4fdba22f8ffe1f5de726e11e949f5e1161:37191
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
Sidebar.jsx @ Sidebar.jsx:1
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
Main.jsx @ Main.jsx:1
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
Routes.jsx @ Routes.jsx:1
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
App.jsx @ App.jsx:1
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
index.js @ index.js:1
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
main.js @ main.js:1
fileEvaluate @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:343
require @ modules-runtime.js?hash=8587d188e038b75ecd27ed2469a52b269e38fb62:238
(anonymous) @ app.js?hash=d83a9ed1037896da45d0c1b0a32c8972cf517af8:670
```

I assume that it's because of missing propTypes in React 16

![image](https://user-images.githubusercontent.com/3371143/30914054-f2ebee9c-a39a-11e7-9882-b393da940dee.png)

**Comment by alpertuna** [comment]
Hello @Kolyasya,
Yep, It is not suitable for latest `react`. I will upgrade it soon.

**Comment by js0n** [comment]
@alpertuna Submitted a PR https://github.com/alpertuna/react-metismenu/pull/34 for the React 16 upgrade which passes CI but not coveralls; any chance to get this reviewed / merged?

It would be nice to be able to pull the package from npm and have React 16 compatibility without depending on a fork.

**Comment by alpertuna** [comment]
Thanks @js0n, now it should be done

**Comment by n0ne** [comment]
Any alternatives to `react-metismenu-router-link`?! Not working with React 16...

**Comment by alpertuna** [comment]
Now It should be fixed by `react-metismenu-router-link@v2.1.0`.
